### PR TITLE
street intersection 2.0

### PR DESCRIPTION
**Pull request by srothst1**
The goal of this pull request is to create a more capable `intersection` component that developers can use to add intersections to their scenes in 3DStreet.

Here are some items that need to be tackled:
- [x] Traffic Signal 
- [x] Stopsign
- [x] Crosswalks
- [x] API documentation
- [x] Crosswalk stencil overlap
- [ ] Fix issue with the stop sign `glTF` model
  - https://github.com/3DStreet/3dstreet/issues/169
  -

### CODE REVIEW

**Comment by srothst1** [comment]
Stop sign issue:
Screen Shot 2022-05-27 at 2 36 05 PM

**Comment by srothst1** [comment]
The following code
```js
  const cw1 = document.createElement('a-entity');
  cw1.setAttribute('position', { x: dimensionsArray[0] / 2, z: 0.04 });
  cw1.setAttribute('rotation', { x: 0, y: 0, z: 180 });
  //cw1.setAttribute('geometry', { height: 18, width: 3}); //THIS LINE CAUSES AN ERROR
  cw1.setAttribute('mixin', 'markings crosswalk-zebra');
  el.appendChild(cw1);
```
produces this error:
```
TypeError: undefined is not an object (evaluating 'mesh.traverse')
```
Any suggestions on how to resolve this? See [e5aeb7d](https://github.com/3DStreet/3dstreet/pull/174/commits/e5aeb7daf37356c6b6ca22badae319118153c1ef) for more context.

**Comment by srothst1** [comment]
Here is a demo scene for experimentation:
```js
  <a-entity
  position="0 0 -35"
  id="left_street"
  street="length: 50"
  streetmix-loader="streetmixStreetURL: https://streetmix.net/kfarr/72/"
></a-entity>

<a-entity
  position="42.4 0 -1"
  rotation = "0 90 0"
  id="perpendicular_street"
  street="length: 55"
  streetmix-loader="streetmixStreetURL: https://streetmix.net/srothst1/7/lapara; showBuildings: false;"
></a-entity>


<a-entity position="0 0 -1" intersection="dimensions: 29.8704 17.9832;sidewalk: 0 6.096 6.096 0;northeastcurb: 4.8768 4.8768;stopsign: 0 0 1 0;trafficsignal: 0 1 0 0;crosswalk: 1 0 0 1"></a-entity>
```

**Comment by kfarr** [comment]
Signalized intersection looking beautiful in its natural habitat 3dtiles.html
image

**Review by kfarr — APPROVED** [review]

File: src/assets.js [hunk 1]
```diff
@@ -5,10 +5,15 @@
...
-  
           <!-- sidewalk props -->
           <img id="wayfinding-map" src="${assetUrl}assets/objects/wayfinding.jpg" crossorigin="anonymous" />
           <a-asset-item id="streetProps" src="${assetUrl}assets/objects/street-props.glb"></a-asset-item>
+          <a-asset-item id="stopsign" src="${assetUrl}assets/signals/stop-sign.glb"></a-asset-item>
+          <a-asset-item id="signal1" src="${assetUrl}assets/signals/signal1.glb"></a-asset-item>
+          <a-asset-item id="signal2" src="${assetUrl}assets/signals/signal2.glb"></a-asset-item>
+          <a-mixin id="signal_left" gltf-part-plus="src: #signal1; part: signal1"></a-mixin>
+          <a-mixin id="signal_right" gltf-part-plus="src: #signal2; part: signal2"></a-mixin>
+          <a-mixin id="stop_sign" gltf-part-plus="src: #stopsign; part: stop-sign"></a-mixin>
```
[the remark this answers is not in the document]
Comment (kfarr) [reply src/assets.js]: done

**Pull request merged**